src/app/files.js

```diff
@@ -0,0 +1,104 @@
+'use strict'
+
+var EventManager = require('../lib/eventManager')
+
+function Files (storage) {
+  var event = new EventManager()
+  this.event = event
+  var readonly = {}
+
+  this.exists = function (path) {
+    // NOTE: ignore the config file
+    if (path === '.browser-solidity.json') {
+      return
+    }
+
+    if (this.isReadOnly(path)) {
+      return true
+    } else {
+      return storage.exists(path)
+    }
+  }
+
+  this.get = function (path) {
+    // NOTE: ignore the config file
+    if (path === '.browser-solidity.json') {
+      return
```

Comment:
I'm not sure whether `localStrage.getItem` returns `null` or `undefined`, but if we return an explicit value these should match.